### PR TITLE
Add `sentry-jdbc` guide.

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Related to https://github.com/getsentry/sentry-java/pull/1784

### CODE REVIEW

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+description: "Learn how to capture the performance of database queries executed with JDBC."
+---
+
+<Note>
+
+Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already.
+
+</Note>
+
+Sentry JDBC integration provides the `SentryJdbcEventListener` for [P6Spy](https://github.com/p6spy/p6spy/) - database activity interceptor - which creates a span for each JDBC statement executed over proxied instance of `javax.sql.DataSource`.
```
Comment (imatwawana) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:13]:
```suggestion
Sentry JDBC integration provides the `SentryJdbcEventListener` for [P6Spy](https://github.com/p6spy/p6spy/)  database activity interceptor, which creates a span for each JDBC statement executed over a proxied instance of `javax.sql.DataSource`.
```

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+```groovy {tabTitle:Gradle}
+implementation 'io.sentry:sentry-jdbc:{{ packages.version('sentry.java.jdbc', '5.3.0') }}'
+```
+
+```scala {tabTitle: SBT}
+libraryDependencies += "io.sentry" % "sentry-jdbc" % "{{ packages.version('sentry.java.jdbc', '5.3.0') }}"
+```
+
+For other dependency managers see the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-jdbc).
```
Comment (imatwawana) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:33]:
```suggestion
For other dependency managers, check out the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-jdbc).
```

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+```properies {tabTitle: application.properties}
+spring.datasource.driver-class-name=com.p6spy.engine.spy.P6SpyDriver
+```
+
+```yaml {tabTitle: application.yml}
+spring.datasource.driver-class-name: com.p6spy.engine.spy.P6SpyDriver
+```
+
+Add `p6spy` prefix to database connection url. For Spring Boot applications:
```
Comment (imatwawana) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:45]:
```suggestion
Add the `p6spy` prefix to the database connection URL. For Spring Boot applications:
```

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+```properies {tabTitle: application.properties}
+spring.datasource.url: jdbc:p6spy:...
+```
+
+```yaml {tabTitle: application.yml}
+spring.datasource.url: jdbc:p6spy:...
+```
+
+P6Spy uses a Java [Service Loader](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) mechanism to register `JdbcEventListener`s. For more configuration options, check [P6Spy Reference Guide](https://p6spy.readthedocs.io/en/latest/).
```
Comment (imatwawana) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:55]:
```suggestion
P6Spy uses a Java [Service Loader](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) mechanism to register `JdbcEventListener`s. For more configuration options, check the [P6Spy Reference Guide](https://p6spy.readthedocs.io/en/latest/).
```

**Review by imatwawana — APPROVED** [review]
Minor wording edits

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+implementation 'io.sentry:sentry-jdbc:{{ packages.version('sentry.java.jdbc', '5.3.0') }}'
+```
+
+```scala {tabTitle: SBT}
+libraryDependencies += "io.sentry" % "sentry-jdbc" % "{{ packages.version('sentry.java.jdbc', '5.3.0') }}"
+```
+
+For other dependency managers see the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-jdbc).
+
+Configure `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:
```
Comment (bruno-garcia) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:35]:
```suggestion
Configure the `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:
```
```suggestion
Configure `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:
```

File: src/platforms/java/performance/instrumentation/jdbc.mdx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+```yaml {tabTitle: application.yml}
+spring.datasource.driver-class-name: com.p6spy.engine.spy.P6SpyDriver
+```
+
+Add `p6spy` prefix to database connection url. For Spring Boot applications:
+
+```properies {tabTitle: application.properties}
+spring.datasource.url: jdbc:p6spy:...
```
Comment (bruno-garcia) [inline src/platforms/java/performance/instrumentation/jdbc.mdx:48]: What about an example instead of `...`?

**Review by bruno-garcia — APPROVED** [review]

File: src/platforms/java/performance/instrumentation/jdbc.mdx (hunk 6 above)
Comment (maciejwalkowiak) [reply src/platforms/java/performance/instrumentation/jdbc.mdx]: 👍

**Comment by maciejwalkowiak** [comment]
Thanks @imatwawana @bruno-garcia for quick review. All suggested changes applied.

**Comment by marandaneto** [comment]
@maciejwalkowiak a problem that I see is that https://sentry-docs-git-fork-maciejwalkowiak-java-jdbc.sentry.dev/platforms/java/performance/instrumentation/jdbc/ is only accessible if the Java platform is selected, if the Spring Boot Guide is selected, we can't see it https://sentry-docs-git-fork-maciejwalkowiak-java-jdbc.sentry.dev/platforms/java/guides/spring-boot/performance/instrumentation/

does it make sense to have such integrations visible also for the Guides?

**Comment by maciejwalkowiak** [comment]
I missed that and assumed it is visible. What's the right approach for making it visible to Spring/Spring Boot guides? Perhaps it would make sense to have a special tag in a frontmatter?

**Comment by marandaneto** [comment]
> I missed that and assumed it is visible. What's the right approach for making it visible to Spring/Spring Boot guides? Perhaps it would make sense to have a special tag in a frontmatter?

I'm not sure either, I believe the current integrations already suffer from this problem, @imatwawana ideas? what would be the best approach? thanks

**Comment by imatwawana** [comment]
> > I missed that and assumed it is visible. What's the right approach for making it visible to Spring/Spring Boot guides? Perhaps it would make sense to have a special tag in a frontmatter?
> 
> I'm not sure either, I believe the current integrations already suffer from this problem, @imatwawana ideas? what would be the best approach? thanks

I **think** the way to fix this is thus:
1. Move the folder **platforms/java/performance/instrumentation** into **platforms/java/common**. This makes the content available to all the Java guides.
2. Update the front matter of all the files in **platforms/java/performance/instrumentation** folder with a supported list to control where the content is visible.

This is the way the Apple content is set up which uses both the global common content for performance as well as content specific to Apple that's common to all of its guides, so it should work.

**Comment by maciejwalkowiak** [comment]
Thanks @imatwawana seems to work!

**Comment by imatwawana** [comment]
> Thanks @imatwawana seems to work with `notSupported`!

Ah good to know!

**Review by marandaneto — APPROVED** [review]

**Pull request merged**